src/subscription_manager/gui/registergui.py

```diff
@@ -1992,6 +2114,24 @@ def _refresh(self, callback):
         except Exception:
             self.queue.put((callback, None, sys.exc_info()))
 
+    def __unregister_consumer(self, consumer_uuid, server_info):
+        # Method to actually do the unregister bits
+        # TUse the last successful connection info
+        self.backend.cp_provider.set_connection_info(**server_info.as_dict())
+        cp = self.backend.cp_provider.get_consumer_auth_cp()
+        managerlib.unregister(cp, consumer_uuid)
```

Comment:
Could we do this in the callback? managerlib.unregister() does a lot, and I'm a little worried about doing that
in a thread.

Maybe even just have the callback emit a 'consumer-unregistered' signal? Not sure how to prevent the next screens [PerformRegister?] pre() from proceeding though, since we do want to wait for unregister to finish before doing the register.